test/test_blockdeque.jl

```diff
@@ -0,0 +1,80 @@
+using BlockArrays, Test
+
+@testset "append!(::BlockVector, vector)" begin
+    @testset for alias in [false, true],
+        compatible in [false, true],
+        srctype in [:BlockVector, :PseudoBlockVector, :Vector]
+
+        dest = mortar([[1, 2, 3], [4, 5]])
+
+        # Create `src` array:
+        if compatible
+            T = Int
+        else
+            T = Float64
+        end
+        if srctype === :BlockVector
+            src = mortar([T[6, 7], T[8, 9]])
+        elseif srctype === :PseudoBlockVector
+            src = PseudoBlockVector(T[6:9;], [3, 1])
+        elseif srctype === :Vector
+            src = T[6:9;]
+        else
+            error("Unknown srctype = $srctype")
+        end
+
+        @test append!(dest, src; alias = alias) === dest
+        @test dest == 1:9
+        src[1] = 666
+        if alias && compatible
+            @test dest[6] == 666
+        else
+            @test dest[6] == 6
+        end
```

Comment:
Add tests (here and below) that make clear the resulting block structure. That is (I think):
```julia
@test dest[Block(1)]  == [1,2,3]
@test dest[Block(2)]  == [4,5,6]
```

Reply:
I added a few tests for the block structure. Actually, `@test dest[Block(2)]  == [4,5,6]` is not always true. Let me know if the block structure makes sense.

Reply:
I think the block structure should be consistent, that is, `append!(::AbstractBlockVector, ::AbstractVector)` should either always add to the last block or add new blocks.

There's a question here what `append!(::AbstractBlockVector, ::AbstractBlockVector)` should do...

Reply:
So, to make things predictable, maybe you'd like something like

`append!(dest, src)`: always mutate the last block (`blocklength(dest)` is constant)

`blockappend!(dest, src)`: always add `blocklength(src)` blocks and avoid copy when make sense

?

Reply:
Yes that's reasonable.